var/spack/repos/builtin/packages/charm/package.py

```diff
@@ -47,63 +50,123 @@ class Charm(Package):
 
     # Support OpenMPI; see
     # <https://charm.cs.illinois.edu/redmine/issues/1206>
-    patch("mpi.patch")
+    # Patch is no longer needed in versions 6.8.0+
+    patch("mpi.patch", when="@:6.7.1")
+
     # Ignore compiler warnings while configuring
     patch("strictpass.patch")
 
+    # Build targets
+    # "target" is reserved, so we have to use something else.
+    variant(
+        "build-target",
+        default="LIBS",
+        values=("charm++", "AMPI", "LIBS"),
+        description="Specify the target to build"
+    )
+
     # Communication mechanisms (choose exactly one)
-    # TODO: Support Blue Gene/Q PAMI, Cray GNI, Cray shmem, CUDA
     variant(
-        'backend',
-        default='mpi',
-        values=('mpi', 'multicore', 'net', 'netlrts', 'verbs'),
-        description='Set the backend to use'
+        "backend",
+        default="mpi",
+        values=("mpi", "multicore", "netlrts", "verbs", "gni",
+                "ofi", "pami", "pamilrts"),
+        description="Set the backend to use"
+    )
+
+    # Compilers (choose exactly one)
+    variant(
+        "compiler",
+        default="gcc",
+        values=("icc", "xlc", "gcc", "clang", "pgcc"),
+        description="Set the compiler"
+    )
+
+    # Fontran compilers (choose exactly one)
+    variant(
+        "fortran-compiler",
+        default="gfortran",
+        values=("gfortran", "flang", "xlf", "ifort", "pgf90"),
+        description="Set the fortran compiler"
     )
 
     # Other options
-    # Something is off with PAPI -- there are build errors. Maybe
-    # Charm++ expects a particular version?
     variant("papi", default=False, description="Enable PAPI integration")
+    variant("syncft", default=False, description="Compile with Charm++ fault tolerance support")
     variant("smp", default=True,
             description=(
                 "Enable SMP parallelism (does not work with +multicore)"))
     variant("tcp", default=False,
             description="Use TCP as transport mechanism (requires +net)")
+    variant("omp", default=False, description="Support for the integrated LLVM OpenMP runtime")
+    variant("pthreads", default=False, description="Compile with pthreads Converse threads")
+    variant("cuda", default=False, description="Enable CUDA toolkit")
+
     variant("shared", default=True, description="Enable shared link support")
+    variant("production", default=True, description="Build charm++ with all optimizations")
+    variant("tracing", default=False, description="Enable tracing modules")
 
-    # Note: We could add variants for AMPI, LIBS, bigemulator, msa, Tau
+    variant("incdir", description="Specify additional include path for compiler")
```

Comment:
why do you need this and the next two?

Reply:
it's there to support building with other libraries or files, the option is also present in the build script that is called at the end so I decided to add it here.

Reply:
All other libraries should come throughout ‘depends_on()’ with optional variant to enable them

Reply:
what i mean is that to this looks like a hack and IMO it does not fit into Spack. All the link-type dependencies should be declared and known so that Spack "concretizes" the graph of packages into a specific fix set of versions/variants/etc.

Reply:
After talking to the team, it seems like these variants can be removed